pkg/controllers/ippool/ippool_controller.go

```diff
@@ -124,6 +124,17 @@ func (r *IPPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 
 	// TODO: Xiaopei's suggestions: is there possibility that IPPool was deleted from nsx store but NSX block subnet was not deleted?
 
+	if obj.Spec.Type == "" {
```

Comment:
sorry for the late awareness.

If spec.Type is set to a value at the beginning, I notice

```
			controllerutil.AddFinalizer(obj, servicecommon.IPPoolFinalizerName)
			if err := r.Client.Update(ctx, obj); err != nil {
				log.Error(err, "add finalizer", "ippool", req.NamespacedName)
				updateFail(r, &ctx, obj, &err)
				return resultRequeue, err
}
```
there will be an add finalizer later.

Will it patch the Type to ippool object? Is this what you want? Have you tested the change locally first?